lib/fail.js

```diff
@@ -27,13 +27,22 @@ module.exports = async (pluginConfig, context) => {
       : undefined
   const repoURL = repoPath && `https://github.com/${repoPath}`
 
+  const failTemplate = pluginConfig.onFailTemplate
   // Override default fail template
-  if (pluginConfig.onFailTemplate) {
-    slackMessage = template(pluginConfig.onFailTemplate, {
+  if (failTemplate) {
+    const variables = {
       package_name,
       repo_path: repoPath,
       repo_url: repoURL
-    })
+    }
+    if (typeof failTemplate === 'object') {
```

Comment:
You can probably replace this with a case statement. Will probably be cleaner.

Example [here](https://github.com/juliuscc/semantic-release-slack-bot/blob/master/lib/template.js)

Reply:
Personally I prefer the `if else` syntax because breaks make switches confusing. But I don't have that strong of a preference so I updated it to use blocked switches. (blocked because the indented syntax is hard to follow)